### PR TITLE
fix: allow to update link url and disallow editing html

### LINKED ISSUE #1057: Update item link creation flow to fit new modal and infos
Todo:
- [x] provided item description should take precedence over injected iframely description
- [x] html integration code should be added by the backend and stripped from the post payload if provided
- [ ] (optional) thumbnails should be generated from the icons to remove the need to ask CDNs and leak user informations

### PR DESCRIPTION
**Pull request by spaenleh**
In this PR:
- move automatic link description to the `extra.description` field
- allow to update a link `url`
- do not allow to update more that the `extra.url` field in updates to prevent injecting into the unsafe `extra.html` prop

fix #1057

### CODE REVIEW

File: src/services/item/plugins/embeddedLink/schemas.ts [hunk 1]
```diff
@@ -15,25 +18,27 @@ export const createSchema = S.object()
...
-//       required: ['embeddedLink']
-//     }
-//   }
-// }
+// schema defining the allowed properties for the link item
+export const updateExtraSchema = S.object()
+  .additionalProperties(false)
+  .prop(
+    ItemType.LINK,
+    // only allow to set url in extra on create
```
Comment (ReidyT) [inline src/services/item/plugins/embeddedLink/schemas.ts:26]: I'm not sure to understand where this constraint is reflected in this schema. It is a description of the next line or an indication that we should allow to set the url in extra on create only ? I think it is the same for the createExtraSchema.

File: src/services/item/plugins/embeddedLink/schemas.ts [hunk 2]
```diff
@@ -15,25 +18,27 @@ export const createSchema = S.object()
...
+    S.object().additionalProperties(false).prop('url', S.string().format('url')).required(['url']),
+  )
+  .required([ItemType.LINK]);
+
+export const getLinkMetadata = {
+  querystring: S.object().additionalProperties(false).prop('link', S.string().format('url')),
```
Comment (ReidyT) [inline src/services/item/plugins/embeddedLink/schemas.ts:32]: Could it be interesting to extract the object schema in a const and reuse it in the three schema ?

File: src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts [hunk 3]
```diff
@@ -189,7 +190,7 @@ describe('Link Item tests', () => {
         ({ app, actor } = await build());
       });
 
-      it('Bad Request for link', async () => {
+      it('Allow to edit link url', async () => {
```
Comment (ReidyT) [inline src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts:193]: So we can update the url ? It is not the opposite of what the comments said ? 😅

**Review by ReidyT — APPROVED** [review]
LGTM 😄 I just let some comments because there is something not clear to me 😅

**Comment by spaenleh** [comment]
@ReidyT Thank you for your review, I tried to make the comments clearer. My intent is to allow only the `extra.url` on creation and update. any other properties are not allowed in the extra payload.

**Comment by ReidyT** [comment]
@spaenleh thanks for this update 👍 ! it is clearer for me now 😄

File: src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts [hunk 4]
```diff
@@ -215,6 +216,34 @@ describe('Link Item tests', () => {
           payload,
         });
 
+        expect(response.statusMessage).toEqual(ReasonPhrases.OK);
+        expect(response.statusCode).toBe(StatusCodes.OK);
+        // TODO: check that the title and description have been updated for the new link
```
Comment (pyphilia) [inline src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts:221]: Todo?

File: src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts [hunk 5]
```diff
@@ -274,8 +303,9 @@ describe('Link Item tests', () => {
           payload,
         });
 
-        expect(response.statusMessage).toEqual(ReasonPhrases.BAD_REQUEST);
-        expect(response.statusCode).toBe(StatusCodes.BAD_REQUEST);
+        expect(response.statusMessage).toEqual(ReasonPhrases.ACCEPTED);
+        expect(response.statusCode).toBe(StatusCodes.ACCEPTED);
+        // TODO: test that the link info has been updated
```
Comment (pyphilia) [inline src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts:308]: Todo

File: src/services/item/service.ts [hunk 6]
```diff
@@ -434,6 +434,8 @@ export class ItemService {
     const item = await itemRepository.get(itemId);
     await validatePermission(repositories, PermissionLevel.Write, actor, item);
 
+    // TODO: if updating a link item, fetch the new informations
```
Comment (pyphilia) [inline src/services/item/service.ts:437]: Todo

**Review by pyphilia — APPROVED** [review]
LTGM, you left some todos I've highlighted them 🥟

File: src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts (hunk 5 above)
Comment (spaenleh) [reply src/services/item/plugins/embeddedLink/test/embeddedLink.test.ts:308]: referenced in #1066

File: src/services/item/service.ts (hunk 6 above)
Comment (spaenleh) [reply src/services/item/service.ts:437]: referenced in #1066

**Pull request merged**